Source/RealmResultsCache.swift

```diff
@@ -120,6 +120,7 @@ class RealmResultsCache<T: Object> {
             guard let section = sectionForOutdateObject(object),
                 let sectionIndex = indexForSection(section) else { continue }
             let index = section.deleteOutdatedObject(object)
+            guard index != -1 else { continue }
```

Comment:
Yes, that would be great. Also guard for `section.indexForOutdatedObject(object)` being different from -1 seems to be in many places. Consider wrapping the whole validation in one method/class.